pkg/daemon/ceph/osd/daemon.go

```diff
@@ -47,7 +50,11 @@ func Run(context *clusterd.Context, agent *OsdAgent, done chan struct{}) error {
 	// set the crush location in the osd config file
 	cephConfig := mon.CreateDefaultCephConfig(context, agent.cluster, path.Join(context.ConfigDir, agent.cluster.Name))
 	cephConfig.GlobalConfig.CrushLocation = agent.location
-
+	// don't set public or cluster addr if prepare only
+	if agent.prepareOnly {
```

Comment:
could this check be done earlier, for example when the command line arguments are parsed?

Reply:
prepareOnly is only needed to reset network address in ceph.conf immediately after it is generated.